src/Compilers/CSharp/Portable/Symbols/NullableAnnotationExtensions.cs

```diff
@@ -16,49 +16,71 @@ internal static class NullableAnnotationExtensions
         /// This uses the contravariant merging rules.
         /// </summary>
-        public static NullableAnnotation Meet(this NullableAnnotation a, NullableAnnotation b)
-        {
-            if (a.IsNotAnnotated() || b.IsNotAnnotated())
-                return NullableAnnotation.NotAnnotated;
-            return (a < b) ? a : b;
-        }
+        public static NullableAnnotation Meet(this NullableAnnotation a, NullableAnnotation b) => (a < b) ? a : b;
 
         /// <summary>
-        /// Check that two nullable annotations are "compatible", which means they could be the same. Return the
-        /// nullable annotation to be used as a result.  This uses the invariant merging rules.
+        /// Return the nullable annotation to use when two annotations are expected to be "compatible", which means
+        /// they could be the same. These are the "invariant" merging rules.
         /// </summary>
-        public static NullableAnnotation EnsureCompatible(this NullableAnnotation a, NullableAnnotation b)
+        public static NullableAnnotation EnsureCompatible(this NullableAnnotation a, NullableAnnotation b) => (a, b) switch
         {
-            if (a.IsOblivious())
-                return b;
-            if (b.IsOblivious())
-                return a;
-            return (a < b) ? a : b;
-        }
+            (NullableAnnotation.Oblivious, _) => b,
+            (_, NullableAnnotation.Oblivious) => a,
+            _ => a < b ? a : b,
+        };
 
         /// <summary>
         /// Merges nullability.
         /// </summary>
-        public static NullableAnnotation MergeNullableAnnotation(this NullableAnnotation a, NullableAnnotation b, VarianceKind variance)
+        public static NullableAnnotation MergeNullableAnnotation(this NullableAnnotation a, NullableAnnotation b, VarianceKind variance) => variance switch
+        {
+            VarianceKind.In => a.Meet(b),
+            VarianceKind.Out => a.Join(b),
+            VarianceKind.None => a.EnsureCompatible(b),
+            _ => throw ExceptionUtilities.UnexpectedValue(variance)
+        };
+
+        /// <summary>
+        /// The attribute (metadata) representation of <see cref="NullableAnnotation.NotAnnotated"/>.
+        /// </summary>
+        public const byte NotAnnotatedAttributeValue = 1;
+
+        /// <summary>
+        /// The attribute (metadata) representation of <see cref="NullableAnnotation.Annotated"/>.
+        /// </summary>
+        public const byte AnnotatedAttributeValue = 2;
+
+        /// <summary>
+        /// The attribute (metadata) representation of <see cref="NullableAnnotation.Oblivious"/>.
+        /// </summary>
+        public const byte ObliviousAttributeValue = 0;
+
+        /// <summary>
+        /// The mapping from NullableAnnotation to the attribute (metadata) representations.
+        /// </summary>
+        public static byte ToAttributeValue(this NullableAnnotation self) => self switch
```

Comment:
>ToAttributeValue [](start = 27, length = 16)

It looks like there is only one call site for this method, it uses only two constant input values and can easily use final constants. Consider removing this method.

Reply:
I prefer to keep it.  I would not be surprised if there are more call sites in the future, or if this changes.

---
In reply to: [274680821](https://github.com/dotnet/roslyn/pull/34909#discussion_r274680821) [](ancestors = 274680821)

Reply:
> I prefer to keep it.  I would not be surprised if there are more call sites in the future, or if this changes.

If we need it, we'll add it. Right now this method is dead code with untested code paths.

---
In reply to: [275049437](https://github.com/dotnet/roslyn/pull/34909#discussion_r275049437) [](ancestors = 275049437,274680821)